### ISSUE TITLE
comment in tox dependencies causes packaging.InvalidRequirement

### ISSUE THREAD

**Issue by jayvdb** [jayvdb/flake8-putty#16]
The flake8 jobs (except on Python 3.2) are currently failing due to something determining there is a syntax error in tox.ini

``` python
Invalid requirement: 'flake8-import-order  # hacking has a similar rule'
Traceback (most recent call last):
  File "/home/travis/build/jayvdb/flake8-putty/.tox/v241/lib/python2.6/site-packages/pip/req/req_install.py", line 78, in __init__
    req = Requirement(req)
  File "/home/travis/build/jayvdb/flake8-putty/.tox/v241/lib/python2.6/site-packages/pip/_vendor/packaging/requirements.py", line 96, in __init__
    requirement_string[e.loc:e.loc + 8]))
InvalidRequirement: Invalid requirement, parse error at "'# hackin'"
```

Also occurring in py27, https://travis-ci.org/jayvdb/flake8-putty/jobs/140301311

This is strange because the last successful build [used pip 6.0.7](https://travis-ci.org/jayvdb/flake8-putty/jobs/123825850#L131) and tox 2.3.1, and the new error is reported by the same version.

The only obvious difference since the last successful build is that virtualenv was 15.0.1 but it is now 15.0.2
Installing virtualenv v13 fixes the problem, but installing 14.0.6 or 15.0.1 doesnt :/

**Comment by The-Compiler** [comment]
I've had the same issue and opened both a tox and a pip issue (no time to find them right now, sorry) - it was agreed that this should be fixed in tox.

virtualenv downloads the newest pip by itself, unless you set `VIRTUALENV_DOWNLOAD=no` in your environment, then it'll use the bundled one. That together with virtualenv 15.0.1 might help.

I ended up writing `requirements.txt` files instead and using `-rrequirements-tests.txt` etc.

**Comment by jayvdb** [comment]
thx for the info.

pywikibot had a very similar problem with pip, which wasnt diagnosed, but I thought might be https://github.com/pypa/pip/issues/3680 , but now I am not so sure .

https://phabricator.wikimedia.org/T135106

**Comment by jayvdb** [comment]
Noting the two issues mentioned above.
https://github.com/pypa/pip/issues/3667
https://bitbucket.org/hpk42/tox/issues/332/
Workaround appears to be virtualenv 15.0.1 and VIRTUALENV_DOWNLOAD=no.  I'll test and merge this first, in a few hours at least.
It does seem like tox is the place to fix this.

**Comment by jayvdb** [comment]
And pypy3.3-5.2-alpha1 requires virtualenv 15.0.2. Other than that it works ok.
https://travis-ci.org/jayvdb/flake8-putty/builds/144342729

**Comment by jayvdb** [comment]
Just noting for my reference, `VIRTUALENV_DOWNLOAD=no` comes from `virtualenv.ConfigOptionParser.update_defaults` calling `self.get_environ_vars`, which loads all 
`VIRTUALENV_*` env vars into the config parser defaults, so this is effectively the same as command line argument `--no-download`

https://github.com/pypa/virtualenv/blob/4b707b8/virtualenv.py#L452
https://github.com/pypa/virtualenv/blob/4b707b8/virtualenv.py#L489

**Comment by jayvdb** [comment]
tox & pip both decided that this is a wont fix.

https://github.com/tox-dev/tox/issues/50#issuecomment-247788593

So the comments must be moved to separate lines, or use old pip, and old packaging, and old .. everything.
